1/main.go

```
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"crypto/rand"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"log"
+	_ "modernc.org/sqlite"
+	"net/http"
+)
+
+const schema = `
+create table if not exists users(
+	user_id integer primary key,
+	name text unique,
+	password text
+);
+create table if not exists sessions(
+	session_id integer primary key,
+	user_id integer references users,
+	token text unique
+);`
+
+func main() {
+	db, err := sql.Open("sqlite", "users.db")
+	if err != nil {
+		log.Fatalln("Failed to open database file", err.Error())
+	}
+	_, err = db.Exec(schema)
+	if err != nil {
+		log.Fatalln("Failed to apply database schema", err.Error())
+	}
+	runServer(db)
+}
+
+func runServer(db *sql.DB) {
+	http.HandleFunc("/register", func(w http.ResponseWriter, r *http.Request) { registerHandler(db, w, r) })
+	http.HandleFunc("/auth", func(w http.ResponseWriter, r *http.Request) { authHandler(db, w, r) })
+	http.HandleFunc("/deauth", func(w http.ResponseWriter, r *http.Request) { deauthHandler(db, w, r) })
+	http.HandleFunc("/profile", func(w http.ResponseWriter, r *http.Request) { profileHandler(db, w, r) })
+	http.ListenAndServe(":8080", nil)
```

A good idea is to check if an error occurred when you start the server instead of ignoring it.

---

If you ignore the error here, you won't know, for example, that the port is already used. The program just won't start without any reason

---

Fixed.